packages/sit-onyx/src/components/OnyxDataGrid/features/renderer.ts

```diff
@@ -19,95 +45,119 @@ const numberFormatter = <TEntry extends DataGridEntry>(
     typeof value === "boolean" ||
     typeof value === "symbol"
   ) {
-    return FALLBACK_RENDER_VALUE;
+    return opts?.fallback ?? FALLBACK_RENDER_VALUE;
   }
 
   const { n } = injectI18n();
 
   // We format the given value as Number. In case it renders as NaN, we replace it with `-`.
   // The typing is incorrect, the `format` method accepts any value
-  return n.value(value as number, "decimal").replace("NaN", FALLBACK_RENDER_VALUE);
+  return n.value(value as number, opts?.format ?? "decimal").replace("NaN", FALLBACK_RENDER_VALUE);
 };
 
-const stringFormatter = <TEntry extends DataGridEntry>(
+export const NUMBER_RENDERER = createTypeRenderer<NumberCellOptions>({
+  header: { component: HeaderCell },
+  cell: {
+    component: ({ metadata, modelValue }) => numberFormatter(modelValue, metadata?.typeOptions),
+    tdAttributes: { class: "onyx-data-grid-number-cell" },
+  },
+});
+
+export type StringCellOptions = {
+  /**
+   * Fallback value to display when the value is undefined.
+   * Defaults to "-" if not provided.
+   */
+  fallback?: string;
+};
+
+export const stringFormatter = <TEntry extends DataGridEntry>(
   value: TEntry[keyof TEntry] | undefined,
+  opts?: StringCellOptions,
 ): string => {
   // using loose "==" here to catch both undefined and null
-  if (value == undefined) return FALLBACK_RENDER_VALUE;
+  if (value == undefined) return opts?.fallback ?? FALLBACK_RENDER_VALUE;
```

Comment:
Can/should we somehow unify this fallback logic? Currently its duplicated for every type renderer